segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -704,11 +718,162 @@ public void readTable(final WireCommands.ReadTable readTable) {
                                               .map(k -> new ByteArraySegment(k.getData().array()))
                                               .collect(Collectors.toList());
         tableStore.get(segment, keys, TIMEOUT)
-                  .thenAccept(values -> connection.send(new WireCommands.TableRead(readTable.getRequestId(), segment, getTableEntries(keys, values))))
+                  .thenAccept(values -> connection.send(new WireCommands.TableRead(readTable.getRequestId(), segment,
+                                                                                   getTableEntriesCommand(keys, values))))
                   .exceptionally(e -> handleException(readTable.getRequestId(), segment, operation, e));
     }
 
-    private WireCommands.TableEntries getTableEntries(final List<ArrayView> inputKeys, final List<TableEntry> resultEntries) {
+    @Override
+    public void getTableKeys(WireCommands.GetTableKeys getTableKeys) {
+        final String segment = getTableKeys.getSegment();
+        final String operation = "getTableKeys";
+
+        if (!verifyToken(segment, getTableKeys.getRequestId(), getTableKeys.getDelegationToken(), operation)) {
+            return;
+        }
+
+        log.info(getTableKeys.getRequestId(), "Fetching keys from {}.", getTableKeys);
+
+        int suggestedKeyCount = getTableKeys.getSuggestedKeyCount();
+        ByteBuffer token = getTableKeys.getContinuationToken();
+
+        byte[] state = null;
+        if (!token.equals(EMPTY_BYTE_BUFFER)) {
+            state = token.array();
+        }
+
+        final AtomicBoolean canContinue = new AtomicBoolean(true);
+        final AtomicInteger keyCount = new AtomicInteger(0);
```

Comment:
I don't think you need this. You can use `keys.size()`

Reply:
yes, removed it.